Lesson 3

khalifdev opened the pull request.

Commit:
Реализовано получение списка товаров из JSON с помощью callback

Commit:
Реализовано получение списка товаров с помощью promise

Commit:
Теперь метод fetchGoods тоже возвращает promise

Commit:
Сделал метод добавления товара в Корзину с выводом списка в console

Commit:
Добавил методы Добавления, Удаления Товаров из Корзины. и вывод корзины

Commit:
Добавил возможность изменения количества товара в Корзине.
Изменил отображение Корзины - сделал дивами

Commit:
Добавил вывод итоговой стоимости Товаров в Корзине

inferal73 commented on script.js:
```diff
@@ -14,7 +37,7 @@ class GoodItem {
                 <p>${this.price || 'Цена по запросу'} руб.</p>
               </div>
-              <button class='addClick' id=${this.id} onclick='listCart.addToCart(this.id)'>
+              <button class='addClick' id=${this.id} onclick='list.addToBasket(this.id)'>
```
события нужно навешивать через addeventlistener

inferal73 commented on script.js:
```diff
@@ -67,40 +91,108 @@ class GoodsList {
+        // уменьшаем количество Товара, если его больше 1
+        let index = this.goods.indexOf(good);
+        if (this.goods[index].count > 1) {
+            good.count--;
```
лучше не использовать унарные операции

inferal73 commented on script.js:
```diff
@@ -28,27 +51,28 @@ class GoodsList {
+    }
+    // метод, отправляющий данные Товара в Корзину для удаления
+    removeFromBasket(id) {
+        listBasket.remove(this.goods[id]);
```
используйте метод find

Commit:
Добавил проверку id перед добавлением/удалением из Корзины.
Убрал унарные операции.
Подключил обработку нажатия на кнопки Купить через addEventListener
